### ISSUE TITLE
Deprecated functions in Resources directory cause collisions

### ISSUE THREAD

**Issue by danFWD** [twigphp/Twig#4180]
Files affected:

- Resources/core.php
- Resources/escaper.php
- Resources/debug.php
- Resources/string_loader.php

Each of these is populated with nothing but globally polluting functions that are all deprecated, making it impossible to run Twig in an environment like say, a Wordpress plugin, that may or may not include another plugin running Twig under a different namespace. 

I understand it may be unrealistic to remove these deprecated functions and hurt backwards compatibility, but perhaps they could be wrapped in a 'function_exists' conditional?

**Comment by fabpot** [comment]
These global functions have been defined in Twig since day 1. They have "just" been moved recently to these files so that we can get rid of them in Twig 4 (have a look there, they're gone).

Are you saying that before their move, it was not an issue?

**Comment by danFWD** [comment]
It wasn’t an issue until I discovered that using Twig 3.10 in a WP plugin with another plugin that uses version 1.44 of Twig caused collisions in the global namespace. 

Perhaps even making these an optional extension like the debug extension would be a plausible path forward. But I see that you are working towards version 4 of Twig, and therefore as long as this is a known issue, then perhaps that is enough for now. Thank you.

**Comment by fabpot** [comment]
Having two major versions of Twig in the same project cannot supported.
The function implementations might be different between the two versions, but only one can be registered. So, it cannot work.

**Comment by danFWD** [comment]
I agree, though I found a hacky workaround for now (until version 4). The two different versions of the Twig class are in separate namespaces, so that works fine. What I had to do was manually delete those deprecated functions from my project. Once I did that, the fatal error went away.

P.S. Also want to throw a **GREAT BIG THANK YOU** out there for your hard work on what has become one of the most important and indispensable packages I use in my day-to-day development.
